### PR TITLE
Remove CHANGELOG.md

### PR DESCRIPTION
**Pull request by ctubbsii**
Remove the CHANGELOG file because it is a pain to manually edit, is easy to be incorrect, and does not add value beyond the value inherent in the source control repository (git log).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,7 +48,7 @@ mvn -Prun-it clean verify
 
 ## Changelog
 
-Refer to changelog for further details [changelog](CHANGELOG.md)
+Refer to the [project repository on GitHub][repo] for further details.
```
Comment (hazendaz) [inline README.md:51]: If removing it, I would remove the entire section.  Its redundant.  This just says look at the repo and links back to itself.

Comment (ctubbsii) [reply README.md:51]: I was thinking if the repo was copied somewhere, the README would stay with it, but I think you're right.

**Comment by hazendaz** [comment]
I think Renovate was reading from there.  Or at least appeared to be doing so.  See https://github.com/hazendaz/base-parent/pull/669.  But I think the change log in general was probably in wrong order where latest should be at top not bottom and depending if it was really reading there or something else since it noted it differently than you have the latest release posted.  Most repos I've worked though don't have a change log like that especially smaller ones like this.  And either some combination of marking it under github releases or milestones are used.  Most often I find myself going right to the releases section anyways on projects to read in more detail and if that isn't used i typically just compare versions and look at history if its needed.

**Comment by hazendaz** [comment]
This is how renovate was showing...

```
Release Notes
revelc/formatter-maven-plugin
[v2.23.0](https://togithub.com/revelc/formatter-maven-plugin/blob/HEAD/CHANGELOG.md#ver-2230)

==========

    Support Eclipse 2023-03 (4.27, JDT 3.33) - requires jdk 11
```

**Comment by hazendaz** [comment]
so guess that answers that.   it was looking into the changelog.

**Comment by hazendaz** [comment]
But I don't see much of a loss there.  It didn't exactly make that jump out.  Most PRs from Renovate do.  That one was hidden in a section that had to be opened to see and besides it links back to the repo as well and typically our only reason for release is almost always just new eclipse versions.  If anything, I'd use the Release section and tag the release with anything of note there in similar way.  I think that can also be automated to pull milestones and do automatigically but I've not done that myself.  I think spotbugs repo works that way.

**Comment by hazendaz** [comment]
See #844 as its very hard to understand and we need something.

**Pull request merged**